cmd/antrea-agent/agent.go

```diff
@@ -328,7 +332,16 @@ func run(o *Options) error {
 	// podUpdateChannel is a channel for receiving Pod updates from CNIServer and
 	// notifying NetworkPolicyController and EgressController to reconcile rules
 	// related to the updated Pods.
-	podUpdateChannel := channel.NewSubscribableChannel("PodUpdate", 100)
+	var podUpdateChannel *channel.SubscribableChannel
+	// externalEntityUpdateChannel is a channel for receiving ExternalEntity updates from ExternalNodeController and
+	// notifying NetworkPolicyController to reconcile rules related to the updated ExternalEntities.
+	var externalEntityUpdateChannel *channel.SubscribableChannel
+	if o.nodeType == config.K8sNode {
+		podUpdateChannel = channel.NewSubscribableChannel("PodUpdate", 100)
+	} else {
+		externalEntityUpdateChannel = channel.NewSubscribableChannel("ExternalNodeUpdate", 100)
+	}
```

Comment:
why don't we use a single channel to keep things simpler? only one of them is ever used anyway and the channel is not strongly typed.

Reply:
@mengdie-song

Reply:
Thanks for the comment. It is true that only one of them will be used at one time, but I am thinking that it is much clearer to have separate channels for Pod and ExternalEntity. 
In this way, it is explicit to show that ExternalNodeController consumes ExternalEntityUpdate channel and others(e.g. NetworkPolicyController, EgressController, cniServer...) consumes PodUpdateChannel. And we will keep the behavior of having only one type of object in one specific channel and do not need to make a name like "EntityUpdate" for a mixed channel. Does it make sense to you? @antoninbas 

The comment reminds me that the name here should be ExternalEntityUpdate instead of ExternalNodeUpdate, will update it.

Reply:
I was suggesting:
```go
var localEndpointUpdateChannel *channel.SubscribableChannel
if o.nodeType == config.K8sNode {
        localEndpointUpdateChannel = channel.NewSubscribableChannel("PodUpdate", 100)
} else {
        localEndpointUpdateChannel = channel.NewSubscribableChannel("ExternalNodeUpdate", 100)
}
```
so the channels would still have different names

I don't feel strongly about this though, so if @tnqn and @jianjuns have already reveiwed and approved that part, feel free to ignore my comment.

Reply:
Thanks Antonin. I prefer to keep two channels internally for now and we can enhance this part later if folks do not like it.